integreat_cms/cms/templatetags/pdf_filters.py

```diff
@@ -30,3 +30,30 @@ def pdf_strip_fontstyles(instance):
         if style := element.attrib.pop("style", None):
             element.attrib["style"] = re.sub(r"font-[a-zA-Z]+:[^;]+", "", style)
     return unescape(tostring(content, with_tail=False).decode("utf-8"))
+
+
+@register.filter
+def pdf_truncate_links(instance, max_chars):
+    """
+    This tag returns the instance with truncated links.
+
+    :param instance: The content object instance
+    :type instance: ~integreat_cms.cms.models.pages.page.Page
+
+    :param max_chars: The maximal length of a link before it will be truncated
+    :type max_chars: int
+
+    :return: The instance with truncated links
+    :rtype: ~integreat_cms.cms.models.pages.page.Page
+    """
+    try:
+        content = fromstring(instance)
+    except ParserError:
+        return instance
+    truncation_postfix = "[...]"
+    truncation_length = max_chars - len(truncation_postfix)
+    for elem, _, _, _ in content.iterlinks():
+        if elem.text and len(elem.text) > max_chars:
+            elem.text = f"{elem.text[:truncation_length]}{truncation_postfix}"
```

Comment:
```suggestion
    for elem, _, _, _ in content.iterlinks():
        elem.text = " ".join(
            Truncator(word).chars(max_chars) for word in elem.text.split(" ")
        )
```
You could also make use of Django's `Truncator` which already handles the truncation postfix on its own...
Also, I suggest to only truncate the specific words in the link text, not the whole text itself.

(this requires `from django.utils.text import Truncator`)